jOOQ/src/main/java/org/jooq/impl/UserImpl.java

```diff
@@ -0,0 +1,21 @@
+package org.jooq.impl;
+
+import org.jooq.User;
+
+/**
+ * A common implementation of the User type
+ *
+ * @author Timur Shaidullin
+ */
+public class UserImpl implements User {
+    private String name;
```

Comment:
The internal representation of a `UserImpl` should be `Name`. The reason for this is the fact that `Name` encodes whether an identifier is quoted or not. In many databases, these are the same identifiers:

- `unquoted`
- `UNQUOTED`

But in most databases, these ones are different:

- `"quoted"` or `[quoted]` or `` `quoted` `` 
- `"QUOTED"` or `[QUOTED]` or `` `QUOTED` ``

Reply:
Sure, you're right.